dlt/destinations/impl/filesystem/layout.py

```diff
@@ -0,0 +1,250 @@
+import re
+from types import TracebackType
+from typing import Any, Dict, List, Optional, Sequence, Set, Type
+
+import pendulum
+from dlt.cli import echo as fmt
+from dlt.common.storages.load_package import ParsedLoadJobFileName
+from dlt.destinations.exceptions import CantExtractTablePrefix, InvalidFilesystemLayout
+from dlt.destinations.impl.filesystem.configuration import (
+    FilesystemDestinationClientConfiguration,
+)
+from typing_extensions import Self
+
+# TODO: ensure layout only has supported placeholders
+SUPPORTED_PLACEHOLDERS = {
+    "schema_name",
+    "table_name",
+    "load_id",
+    "file_id",
+    "ext",
+    "curr_date",
+    "year",
+    "month",
+    "day",
+    "hour",
+    "minute",
+    "dow",
+    "timestamp",
+}
+
+SUPPORTED_TABLE_NAME_PREFIX_PLACEHOLDERS = ("schema_name",)
+
+
+class ExtraParams:
+    def __init__(
+        self,
+        config: FilesystemDestinationClientConfiguration,
```

Comment:
it would be good if we have no dependencies on config in this module. just pass layout and extra_placeholders directly